src/pages/gateway/hooks.md

```diff
@@ -0,0 +1,342 @@
+---
+title: Hooks transform
+description: Learn how to use the Hooks transform to add hooks before and after querying your mesh.
+---
+
+# Hooks transform
+
+The Hooks transform allows you to invoke a composable [local or remote](#local-vs-remote-functions) function on a targeted node.
+
+Some use cases for the `HooksTransform` include:
+
+-  Authenticating a user before all operations
+
+-  Publish events once all operations are executed
+
+-  Checking for `storeCode` in headers before executing `availableStores` query (Adobe Commerce)
```

Comment:
This was just an example I used since I dint have a concrete one ATM. You can remove this from the sample use cases list, since it isn't a valid one

Reply:
@revanth0212 which use case should be removed, line 14 or line 16?

Reply:
Please remove the `Checking for `storeCode` in headers` usecase